src/types.ts

```diff
@@ -89,3 +92,68 @@ export interface AmplitudeReactNativeModule {
     eventUploadThreshold: number,
   ): Promise<boolean>;
 }
+
+/**
+ * Strings that have special meaning when used as an event's type
+ * and have different specifications.
+ */
+export enum SpecialEventType {
+  IDENTIFY = '$identify',
+  GROUP_IDENTIFY = '$groupidentify',
+}
+
+export type BaseEvent = {
+  eventType: Exclude<string, SpecialEventType>;
+  userId?: string;
+  deviceId?: string;
+  eventProperties?: PropertiesObject;
+};
+
+export type IdentifyEvent = {
+  eventType: SpecialEventType.IDENTIFY;
+  userId?: string;
+  deviceId?: string;
+  userProperties: IdentifyPayload;
+};
+
+export type GroupIdentifyEvent = {
+  eventType: SpecialEventType.GROUP_IDENTIFY;
+  userId?: string;
+  deviceId?: string;
+  groupType: string;
+  groupName: string | string[];
+  groupProperties: IdentifyPayload;
+};
```

Comment:
Can you please change these types to interfaces and have them extend BaseEvent similar to Amplitude-Node. It will make the `Event` type always include eventProperties and make usage easier in middleware.

```
export interface IdentifyEvent extends BaseEvent {
  eventType: SpecialEventType.IDENTIFY;
  userProperties: IdentifyPayload;
}

export interface GroupIdentifyEvent extends BaseEvent {
  eventType: SpecialEventType.GROUP_IDENTIFY;
  groupType: string;
  groupName: string | string[];
  groupProperties: IdentifyPayload;
}
```

Reply:
Done